### DISCUSSION TITLE
複数1:N関係のテーブルでCriteria APIのassociateを利用すると、重複データが取得されてしまいます。

### DISCUSSION THREAD

**Discussion by gikeihi** [domaframework/doma#890] [Q&A] [answered]
t_aとt_bは1:nの関係、更にt_aとt_cも1:Nの関係です.

下記のような使い方で
```
      .associate(ta, tb, (a, b) -> a.getTbs().add(b))
      .associate(ta, tc, (a, c) -> a.getTcs().add(c))
```
bとc両方とも重複データが取得されています。
使い方が間違っていますか？

**Comment by nakamura-to** [comment, marked as answer]
仕様としては重複が発生しないのが正しいです。
提示いただいたコード断片は特に問題ないように見えます。

重複チェックは`@Id`が付与されたプロパティを使って行っていますが、bやcのエンティティ定義は正しいでしょうか？

もし可能であれば再現するプロジェクトなどを共有いただけると助かります。

**Reply by gikeihi** [reply]
早速のご回答ありがとうございました。
提出情報が不足で申し訳ございません。
簡単にプロジェクトを作成しましたので、ご確認をお願いします。

https://github.com/gikeihi/doma-demo.git
データベース：h2, メモリモード
DDL:
src/main/resources/schema.sql
src/main/resources/data.sql

プロジェクトをcloneして、プロジェクトのフォルダーに下記コマンドで起動できます。
`./gradlew bootRun`
その後ブラウザで`http://localhost:8080/`を開くと結果を見れます。

**Reply by nakamura-to** [reply]
プロジェクトの共有、ありがとうございます。確認します。

**Reply by nakamura-to** [reply]
Domaの不具合だということがわかりました。
今週末に修正して新しいバージョンもリリースしたいと思います。

**Reply by nakamura-to** [reply]
修正したバージョンをリリースしました。こちらでお試しください。
https://github.com/domaframework/doma/releases/tag/2.53.1

**Reply by gikeihi** [reply]
早速のご対応、ありがとうございました。
最新版を試します。
